### PR TITLE
HRINT-2698 - Highlighting Query Results - Basic

### PR DESCRIPTION
**Pull request by Drorlev**
Highlighting Query Results - Basic

### CODE REVIEW

**Review by ml054 — APPROVED** [review]

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
+package net.ravendb.demo.textSearch.highlightQueryResultsBasics;
+//region Usings
+
+import net.ravendb.client.documents.indexes.AbstractIndexCreationTask;
+import net.ravendb.client.documents.indexes.FieldIndexing;
+import net.ravendb.client.documents.indexes.FieldStorage;
+import net.ravendb.client.documents.indexes.FieldTermVector;
+import net.ravendb.client.documents.queries.highlighting.Highlightings;
+import net.ravendb.client.documents.session.IDocumentSession;
+
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java:10]: better to remove this line break

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import static net.ravendb.client.documents.queries.Query.index;
+
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java:12]: better to remove this line break

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
+package net.ravendb.demo.textSearch.highlightQueryResultsBasics;
+//region Usings
+
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java:3]: better to remove this line break

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+                "* Query the 'Employees' collection using the defined index for documents that contain the specified term.",
+                "",
+                "* In this example,<br> we are searching for documents containing the term 'sales' within their 'Notes' field.<br>",
+                "  In addition, we request the highlighted fragments.",
+                "",
+                "* `highlight()`<br>",
+                "   * Specify the index field for which to get text fragments with the term highlighted.<br>",
+                "   * fragmentLength - the maximum length for a fragment (length must be >= 18).<br>",
+                "   * fragmentCount - used to limit the number of matching fragments returned per document.<br>",
+                "   * Provide an 'out Highlightings' param for the fragments' results.",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json:170]: No `out` in Java 
So just change to:
=>
Provide a 'Highlightings' param for the fragments' results

**Review by Danielle9897 — CHANGES_REQUESTED** [review]

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java (hunk 1 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java:10]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java (hunk 2 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java:12]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java (hunk 3 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java:3]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json (hunk 4 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json:170]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    public class EmployeesDetails extends AbstractIndexCreationTask {
+    //endregion
+        public EmployeesDetails() {
+            //region Step_2
+            map = "docs.Employees.Select(employee => new {\n" +
+                "    Notes = employee.Notes[0]\n" +
+                     // employee.Notes is a string array,
+                     // indexing only the first element for this example
+                "})";
+                //endregion
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java:28]: pls align `//endregion` with `//region`

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        //region Demo
+        List<Employee> employeesResults;
+
+        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
+            //region Step_4
+            employeesResults = session.query(Employee.class, index("EmployeesDetails"))
+                .highlight("Notes", fragmentLength, fragmentCount, notesHighlightings)
+                .search("Notes", "sales")
+                .toList();
+                 //endregion
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java:54]: pls align `//endregion` with `//region`

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+        "* Fragments count<br>",
+        "* Tags marking the terms within the fragments<br>",
+        " (see the Customized Demo )</p>"
+    ],
+
+    "Assets": [
+        {
+            "Title": "FTS - Static Index Demo",
+            "Type": "demo",
+            "Url": "/java/text-search/fts-with-static-index-single-field"
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json:22]: pls return all spaces to how it was, 
we talked about that in the very beginning... 
the current convention for all metadata files in the demo is:


image

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+                "* Query the 'Employees' collection using the defined index for documents that contain the specified term.",
+                "",
+                "* In this example,<br> we are searching for documents containing the term 'sales' within their 'Notes' field.<br>",
+                "  In addition, we request the highlighted fragments.",
+                "",
+                "* `highlight()`<br>",
+                "   * Specify the index field for which to get text fragments with the term highlighted.<br>",
+                "   * fragmentLength - the maximum length for a fragment (length must be >= 18).<br>",
+                "   * fragmentCount - used to limit the number of matching fragments returned per document.<br>",
+                "   * Provide an 'Highlightings' param for the fragments' results.",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json:172]: Provide an => Provide a

**Review by Danielle9897 — CHANGES_REQUESTED** [review]

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json (hunk 7 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json:22]: done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json (hunk 8 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/metadata.json:172]: done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java (hunk 6 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java:54]: done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java (hunk 5 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/highlightQueryResultsBasics/HighlightQueryResultsBasics.java:28]: done

**Review by Danielle9897 — APPROVED** [review]

**Comment by Danielle9897** [comment]
@ml054 this can be merged

**Pull request merged**